### PR TITLE
Add Box utility functions

### PR DESCRIPTION
**Pull request by ssr-yuki**
This PR adds the following utility functions for `Box` struct:
- std::string boxToString(const Box& box)
- Box makeBoxFromTwoPoints(const std::vector<geometry_msgs::Point32>& points)
- Box makeBoxFromXMLRPC(const XmlRpc::XmlRpcValue& box_xmlrpc, const std::string& full_param_name)
- Box makeBoxFromString(const std::string& box_string, const Box& last_box)
- Box padBox(const Box& polygon, double padding)

### CODE REVIEW

File: src/box_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  if (points.size() == 2)
+  {
+    box.min.x = std::min(points[0].x, points[1].x);
+    box.min.y = std::min(points[0].y, points[1].y);
+    box.min.z = std::min(points[0].z, points[1].z);
+
+    box.max.x = std::max(points[0].x, points[1].x);
+    box.max.y = std::max(points[0].y, points[1].y);
+    box.max.z = std::max(points[0].z, points[1].z);
+  }
```
Comment (OTL) [inline src/box_utils.cpp]: I think it seems better to return some error information if the input vector size is not two.
How about adding ROS_ERROR or ROS_WARN?

File: src/box_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                               "i.e. [[x1, y1, z1], [x2, y2, z2]], but this spec is not of that form");
+    }
+
+    points[i].x = getNumberFromXMLRPC(point_xmlrpc[0], full_param_name);
+    points[i].y = getNumberFromXMLRPC(point_xmlrpc[1], full_param_name);
+    points[i].z = getNumberFromXMLRPC(point_xmlrpc[2], full_param_name);
+  }
+
+  Box box = makeBoxFromTwoPoints(points[0], points[1]);
+  return box;
```
Comment (OTL) [inline src/box_utils.cpp]: I think removing the `box` is better to reduce code.

```suggestion
  return makeBoxFromTwoPoints(points[0], points[1]);
```

File: src/box_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+  // convert vvf into points.
+  if (vvf.size() != 2)
+  {
+    ROS_WARN("You must specify just two points to define a box");
+    return last_box;
+  }
+
+  std::vector<geometry_msgs::Point32> points;
+  points.resize(vvf.size());  // points.size() == 2
```
Comment (OTL) [inline src/box_utils.cpp]: How about using a constructor to initialize the size?

```suggestion
  std::vector<geometry_msgs::Point32> points(vvf.size());  // size must be 2
```

Comment (ssr-yuki) [reply src/box_utils.cpp]: Looks good to me.
Using the same representation, I am going to fix here:
https://github.com/smilerobotics/laser_filters/blob/7bb13b8c9c22adbcc321e91455e83c33a8347a01/src/box_utils.cpp#L89

File: src/box_utils.cpp (hunk 1 above)
Comment (ssr-yuki) [reply src/box_utils.cpp]: You are right.
I have re-designed the function in [here](https://github.com/smilerobotics/laser_filters/pull/5/commits/d58c6c97442f0a2c48e115fe6ac25c6ab4726fa5).

**Review by OTL — APPROVED** [review]

**Pull request merged**